src/task.h

```diff
@@ -127,12 +127,25 @@ task_thread_create(const char            *name,
                    platform_thread       *thread);
 
 // Register the calling thread, allocating scratch space for it
+#define task_register_this_thread(ts, scratch_size)                            \
+   task_register_thread((ts), (scratch_size), __FILE__, __LINE__, __FUNCTION__)
```

Comment:
Changed this and on L141 below to become caller-macros, so I can pass-down code location tags to the lower level function, for improved diagnostics.